pandas/core/groupby/generic.py

```diff
@@ -834,9 +833,10 @@ def _wrap_output(self, output, index, names=None):
             return Series(output, index=index, name=name)
 
     def _wrap_aggregated_output(self, output, names=None):
-        return self._wrap_output(output=output,
-                                 index=self.grouper.result_index,
-                                 names=names)
+        result = self._wrap_output(output=output,
+                                   index=self.grouper.result_index,
+                                   names=names)
+        return self._reindex_output(result)._convert(datetime=True)
```

Comment:
What is the point of convert here?

Reply:
I believe it tries to convert object columns that look like dates to datetime. I saw that `DataFrameGroupBy` does this in `wrap_aggregated_output`, but not in `wrap_applied_output`, so I wanted to stay consistent and do it in the same way for `SeriesGroupBy`. On second thought, I am not sure whether to keep it or not. What do you think?

Reply:
Hmm alright then. I guess let's keep for now. It does seem strange at first glance but can leave it to a follow up to clean that up